src/coreclr/jit/lower.cpp

```diff
@@ -3698,11 +3698,28 @@ GenTree* Lowering::LowerDirectCall(GenTreeCall* call)
     switch (accessType)
     {
         case IAT_VALUE:
+        {
+            bool avoidRelativeCall = false;
+#ifdef TARGET_ARM64
+            if (comp->eeGetRelocTypeHint(addr) != IMAGE_REL_ARM64_BRANCH26)
```

Comment:
Looks like technically what I need here is "give me approximate location in memory where current method will be saved to" and I'll subtract this `addr` from it and if the difference is larger than let's say 127Mb -> emit direct call via reg. 127 (or smaller) just because we don't know where the `call` instruction will land in the end at this point.

I know that in theory